frameworklauncher/src/main/java/com/microsoft/frameworklauncher/applicationmaster/Node.java

```diff
@@ -40,6 +40,29 @@ public Node(String host, Set<String> labels, ResourceDescriptor totalResource, R
     this.requestedResource = ResourceDescriptor.newInstance(0, 0, 0, 0L);
   }
 
+  // Compare two node's AvailableResource,  order is Gpu, Cpu, Memory
+  @Override
+  public int compareTo(Node arg0) {
+    if (this.getAvailableResource().getGpuNumber() > arg0.getAvailableResource().getGpuNumber())
```

Comment:
>getAvailableResource() [](start = 13, length = 22)

You can getAvailableResource only one time, and then reuse it. Instead of call getAvailableResource every time,